homeassistant/components/zcs_azzurro_inverter/__init__.py

```diff
@@ -0,0 +1,51 @@
+"""The zcs_azzurro integration."""
+from __future__ import annotations
+
+import logging
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import DOMAIN, SCHEMA_CLIENT_KEY, SCHEMA_FRIENDLY_NAME, SCHEMA_THINGS_KEY
+from .zcs_azzurro_api import ZcsAzzurroApi
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up zcs_azzurro from a config entry."""
+
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][entry.entry_id] = ZcsAzzurroApi(
+        client=entry.data[SCHEMA_CLIENT_KEY],
+        thing_serial=entry.data[SCHEMA_THINGS_KEY],
+        name=entry.data[SCHEMA_FRIENDLY_NAME],
+    )
```

Comment:
hass.data should only be set once you can confirm that you can authenticate and connect. You probably should also be using ConfigEntryNotReady and ConfigEntryAuthFailed Exceptions

Reply:
changed. Same as above, scaffold put it here